### DISCUSSION TITLE
'Replace all' prompts to replace a word twice

### DISCUSSION THREAD

**Discussion by u-fischer** [WinEdt-Team/WinEdt-Support#63] [1.2 Bugs and Issues]
To reproduce:

* put the cursor at the *begin* of a word (this is relevant, it does not happen if one start the selection from the end)
* select the word with shift + right arrow
* press ctrl+R to open the replace dialog, the word is in the "search for" field
* put something in the replace field
* trigger `replace all`
* press no for the first hit, yes for the following up hits

Winedt then returns at the end to the first occurence and asks again if it should be replaced. E.g. in the following screenshots there are two `blub` in the text but Winedt reports that it replaced 1 of 3. 

![image](https://github.com/user-attachments/assets/741cef8c-f237-4fea-8e7e-f17585c58f4f)

![image](https://github.com/user-attachments/assets/078b6f85-25ce-4cd7-8590-cf7abf8e5618)

**Comment by Alex-Simonic** [comment]
This looks like a bug. I will investigate and report back when I start working on WinEdt again (a month or two).l

**Reply by Alex-Simonic** [reply]
Can you confirm that the problem only occurs if Cyclic Search AND Entire Scope options are both enabled?

**Comment by u-fischer** [comment]
> Can you confirm that the problem only occurs if Cyclic Search AND Entire Scope options are both enabled?

Sorry I overlooked that question. Yes it happens only with *Cyclic Search* and  *Entire Scope*, *Cyclic Search* and *From Cursor* seems to be fine. That could mean that the problem is older as I normally have Cyclic search enabled.

**Reply by Alex-Simonic** [reply]
This has now been fixed and extensively tested (as much as I can do it alone). The problem was introduced in an early WinEdt 11.

There is very little interest in beta testing of new WinEdt releases and sometimes good intentions backfire (especially without enough quality feedback during the beta phase).
